### PR TITLE
[Metric] Add quadratic kappa

### PR DESCRIPTION
**Pull request by sxjscience**
*Description of changes:*

Add Quadratic Kappa as a metric. Reference: https://www.kaggle.com/aroraaman/quadratic-kappa-metric-explained-in-5-simple-steps

Quadratic Kappa is designed for rating predictions, when predicting 1 with 2 will have smaller penalty than predicting 1 with 5.

FYI @sallypannn

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1104-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1104/1/index.html

**Comment by szha** [comment]
Job PR-1104-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1104/2/index.html

File: core/src/autogluon/core/metrics/classification_metrics.py [hunk 1]
```diff
@@ -331,4 +333,35 @@ def confusion_matrix(solution, prediction, labels=None, weights=None, normalize=
...
+        Shape (#samples,)
+    y_pred
+        Shape (#samples, #class) or (#samples,)
+
+    Returns
+    -------
+    score
+        scalar score
+    """
+    labels = None
```
Comment (jwmueller) [inline core/src/autogluon/core/metrics/classification_metrics.py:360]: shouldn't `labels` be optional arg here?

**Review by jwmueller — APPROVED** [review]
I think `labels` is supposed to be optionl arg of the metric? Otherwise it can be deleted

**Comment by szha** [comment]
Job PR-1104-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1104/3/index.html

**Pull request merged**